### DISCUSSION TITLE
Pose estimation on animal-centered videos

### DISCUSSION THREAD

**Discussion by antortjim** [talmolab/sleap#1304] [General]
Dear SLEAP devs

I want to perform pose estimation on collections of videos where the task of finding each separate animal (fly) is already performed by an external program (idtracker.ai). idtracker.ai provides the bounding box and I want to feed the resulting animal-centered video to SLEAP for pose estimation.
I could either use 1) a bottom-up model, or 2) a top-down model where the centroid step is skipped (because it's always the center of the frame by definition). I am more inclined to use 2) a top-down model because I have used them in the past in other projects, but I don't know if it's possible to skip the centroid detection step. So maybe I should go for 1)?
One important aspect of these videos is that while the frame always contains the same animal in the center, it is not always that no other body part of another animal appears, because another animal can come close enough to the focal one to partially appear in the bounding box.

Thanks!
Antonio

**Comment by antortjim** [comment]
Sorry, I missed this nice illustration from https://sleap.ai/guides/troubleshooting-workflows.html 
![image](https://user-images.githubusercontent.com/13869571/236646062-f98f6126-8bc4-43f2-9bb5-fe56fef7d037.png)

and I guess I need to run a top-down model.

However, I would still prefer to skip the centroid detection step, because I already know the output should always be the center of the frame. I assume that I indeed cannot run a single animal model because there will for sure be some body parts in some frames belonging to a second or even third animal. If I did, I would run into the risk of the model catching the right body part but from the wrong animal.

Thanks again!
Antonio

**Comment by antortjim** [comment]
Coming back to this discussion just to elaborate on what I decided to do. SLEAP inference is very very fast, and my frames are only 100x100 in size, so I believe the first model runs extremely fast, and is very accurate anyway. So I confirm the Top-Down 2-step model is working really well in my use case :)

**Reply by talmo** [reply]
Thanks for updating us @antortjim! Apologies we couldn't get to your question sooner but thanks for circling back with the solution.

I'll just add that technically for a pre-centered video you could probably get away with using a single instance model so you don't incur the overhead of the centroid model (which idtracker replaces in this case).

**Reply by antortjim** [reply]
Hi @talmo ! Thank you for getting back so quickly. One of the best things of using SLEAP is the excellent support :heart: 
Cool, so then I can use a single instance model even if body parts of neighboring animals could still appear around the focal animal?
Thank you so much!
A

**Reply by talmo** [reply]
Yep! The model will learn implicitly to ignore animals in the periphery provided that it has a large enough max receptive field. For 100x100 images i think a max stride of 16 will be plenty.

Let us know how it works or if you have any questions!

**Reply by antortjim** [reply]
Actually, I was wondering how you chose the receptive field. My rule of thumb is that a human should be able to tell what body part one is looking at by only looking within the square. But maybe that lower bound is a bit too optimistic, and I should make it bigger, not sure. Thank you!

**Reply by talmo** [reply]
That's definitely the minimum, but if you're not getting good results, try bumping it up to max stride of 32. I'd try to increase the number of filters once you've crossed that threshold.
